### PR TITLE
[NEW] Allow registering by REG_TOKEN environment variable

### PR DESCRIPTION
**Pull request by geekgonecrazy**
## Proposed changes (including videos or screenshots)

You can provide the REG_TOKEN environment variable containing a registration token and it will automatically register to your cloud account.  This simplifies the registration flow

### CODE REVIEW

File: app/cloud/server/index.js [hunk 1]
```diff
@@ -34,6 +37,22 @@ Meteor.startup(function() {
...
+
+	if (process.env.REG_TOKEN && process.env.REG_TOKEN !== '' && !workspaceRegistered) {
+		try {
+			SystemLogger.info('REG_TOKEN Provided. Attempting to register');
+
+			if (!connectWorkspace(process.env.REG_TOKEN)) {
+				throw new Error('Couldn\'t register with token.  Please make sure token is valid or hasn\'t already been used');
+			}
+
+			console.log('Successfully registered with token provided by REG_TOKEN!');
```
Comment (KevLehman) [inline app/cloud/server/index.js:51]: Use SystemLogger here

File: app/cloud/server/index.js [hunk 2]
```diff
@@ -34,6 +37,22 @@ Meteor.startup(function() {
 			job: syncWorkspace,
 		});
 	});
+
+	const { workspaceRegistered } = retrieveRegistrationStatus();
+
+	if (process.env.REG_TOKEN && process.env.REG_TOKEN !== '' && !workspaceRegistered) {
```
Comment (KevLehman) [inline app/cloud/server/index.js:43]: Maybe we can change this to test for negative to avoid nesting the if 👀

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Review by felipe-menelau — APPROVED** [review]
There a github warning regarding a Forbidden non-null assertion on the diff, but I'm guessing it's unrelated?
Otherwise LGTM!

File: app/cloud/server/index.js (hunk 1 above)
Comment (geekgonecrazy) [reply app/cloud/server/index.js:51]: Which variation?  I want this one to always log

File: app/cloud/server/index.js (hunk 2 above)
Comment (graywolf336) [reply app/cloud/server/index.js:43]: @KevLehman can you make a suggestion here with an example of what you mean?

Comment (KevLehman) [reply app/cloud/server/index.js:43]: It's something like:
```
if (!process.env.REG_TOKEN || process.env.REG_TOKEN.trim() === '' || workspaceRegistered) return;
try { ... }
```

File: app/cloud/server/index.js (hunk 1 above)
Comment (pierre-lehnen-rc) [reply app/cloud/server/index.js:51]: On the default log level we only log errors. The next level is `SystemLogger.info`

Comment (graywolf336) [reply app/cloud/server/index.js:51]: So then console.log is the only option for logging 100% of the time without the log being a false error?

Comment (geekgonecrazy) [reply app/cloud/server/index.js:51]: Definitely seems this way

**Pull request merged**